Lib/test/test_linecache.py

```diff
@@ -181,6 +188,37 @@ def test_checkcache(self):
                 self.assertEqual(line, getline(source_name, index + 1))
                 source_list.append(line)
 
+    def test_checkcache_oserror(self):
+        linecache.clearcache()
+        _ = linecache.getlines(FILENAME)
+        self.assertTrue(_)
+        self.assertEqual(1, len(linecache.cache.keys()))
+
+        with support.swap_attr(os, 'stat', raise_oserror):
+            # pop all cache
+            _ = linecache.checkcache()
+            self.assertEqual(0, len(linecache.cache.keys()))
+
+    def test_updatecache_oserror(self):
+        linecache.clearcache()
+        source_name = os_helper.TESTFN
+        self.addCleanup(os_helper.unlink, os_helper.TESTFN)
+
+        with open(source_name, 'w', encoding='utf-8') as source:
+            source.write(SOURCE_4)
+        _ = linecache.getlines(source_name)
+        self.assertEqual(1, len(linecache.cache.keys()))
+
+        with support.swap_attr(os, 'stat', raise_oserror):
```

Comment:
As above - delete/modify the file instead of mocking os.stat.

Reply:
As shown above, I write delete/modify test.